### DISCUSSION TITLE
aiida_cryspy

### DISCUSSION THREAD

**Discussion by nim-hrkn** [Tomoki-YAMASHITA/CrySPY#26] [General]
Prototype that works RS with lammps.
https://github.com/nim-hrkn/aiida_cryspy

**Comment by nim-hrkn** [comment]
内部構造がよく分からないので、pkl_dataを使うように直します。

**Comment by yw-fang** [comment]
Perfect! I was also thinking to implement a plugin to aiida. Your implementation looks good!

**Reply by nim-hrkn** [reply]
Thanks,
I have a trouble. I can't follow the changes of data in pkl_data completely, and I can't mimic them all. 
It will be easier to hold the original structure of the program,. It means that the data in pkl_data as well as 'stat' could be inputs/outputs  of workchains and calculations.
I must design the workhains again to hold the original structure of the program. 
The final form of aiida_cryspy would be similar to my first implementation.

**Reply by yw-fang** [reply]
Oops, I missed so many discussions these days when I was writing a proposal! It seems that both of you have solved this problem.

**Comment by nim-hrkn** [comment]
input/outputの繋がりが不完全だが、設計を替えてAiiDA制御でEAが動くようなった。

TODO:

- initial structureなど結晶構造dict
- stat
- ea_*

がaiidaのinput/outputとする。

- dataframe pluginはあるので使う。
https://github.com/janssenhenning/aiida-dataframe
- 結晶構造リストtrajectoryは使えない。Data pluginを作った方が良い。
- ConfigParser)()は,_sectionでdictになるのでdictで良い。

**Comment by nim-hrkn** [comment]
jobs.tmp_running, tmp_queueingはファイルにsaveしてないですよね。

**Reply by Tomoki-YAMASHITA** [reply]
実行中のテンポラリ変数なので，ファイルには保存してないですね．
このあたりaiida用に書き換えないといけないかもしれませんね．

**Comment by nim-hrkn** [comment]
状態変化を書き込むsubroutineが奥にあって追うのが大変だ。

**Comment by nim-hrkn** [comment]
aiida_cryspyをupdateしました。

https://github.com/nim-hrkn/CrySPY
のargument並び順をある程度きれいにしました。
LAQAはまだ使えません。
